guacamole-ext/src/main/java/org/apache/guacamole/properties/UriGuacamoleProperty.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.guacamole.properties;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import org.apache.guacamole.GuacamoleException;
+import org.apache.guacamole.GuacamoleServerException;
+
+/**
+ * A GuacamoleProperty whose value is a URI.
+ */
+public abstract class UriGuacamoleProperty implements GuacamoleProperty<URI> {
```

Comment:
With the competing camps of applying camelcase to variable/class/function names containing abbreviations, the Guacamole codebase has historically gone the `URIGuacamoleProperty` route rather than the `UriGuacamoleProperty` route. I would suggest doing that here for consistency's sake.

Reply:
Fair enough, I'll rename it.

Reply:
Renamed to `URIGuacamoleProperty`.